packages/frontend/src/features/sqlRunner/store/sqlRunnerSlice.ts

```diff
@@ -93,7 +93,10 @@ export const sqlRunnerSlice = createSlice({
             state,
             action: PayloadAction<ResultsAndColumns>,
         ) => {
-            if (!action.payload.results || !action.payload.columns) {
+            if (
+                action.payload.results.length === 0 ||
```

Comment:
besides `length` do we need to also check if they don't exist at all?

Reply:
from the type that is there they should always exist,checking where they are coming from to understand if that's a possibility